### PR TITLE
Corrected inaccurate build instructions and vague points in install instructions

### PR DESCRIPTION
**Pull request by Songtech-0912**
##  PR Description

Signed-off-by: Jacky Song <dixinlily@hotmail.com>

* The original install instructions for Linux systems had the incorrect instruction of `cmake -G "Unix Makefiles" -Duse_ffmpeg=loaded ../tenacity` where it should've actually been `cmake -G "Unix Makefiles" -Duse_ffmpeg=loaded ..`
* As well as fixing this, I also added an alternate instruction for running `make` to reduce processor resource use and to explicitly specify the different options of compiling (which may be confusing for beginners)

- [x] I have signed off my commits using `-s` or `Signed-off-by`
- [x] I made sure the code compiles on my machine (this change is not related to code)
- [x] I made sure there are no unnecessary changes in the code
- [x] I made sure the title of the PR reflects the core meaning of the issue you are solving
- [x] I made sure the commit message(s) contain a description and answer the question "Why do those changes fix that particular issue?" or "Why are those changes really necessary as improvements?"

### CODE REVIEW

File: BUILDING.md [hunk 1]
```diff
@@ -125,16 +125,25 @@ cmake -GXCode -T buildsystem=1 -Duse_mad="off" -Duse_id3tag=off ../tenacity
     ```
 
 2. Configure Tenacity using CMake:
-   ```
+   ```bash
    $ mkdir build && cd build
-   $ cmake -G "Unix Makefiles" -Duse_ffmpeg=loaded ../tenacity
+   $ cmake -G "Unix Makefiles" -Duse_ffmpeg=loaded ..
```
Comment (fossdd) [inline BUILDING.md:131]: This doesn't work. `build` isn't in the same directory as the git repo.

**Review by fossdd — CHANGES_REQUESTED** [review]
nice PR

File: BUILDING.md (hunk 1 above)
Comment (Songtech-0912) [reply BUILDING.md:131]: Oops, my bad. I'll correct that. I just instinctively got used to cd-ing into a repo once I cloned it so I didn't get why it didn't work as intended :(

**Comment by Songtech-0912** [comment]
I've fixed the changes. Unfortunately, I accidentally forgot to sign the second commit. Merge anyway?

**Comment by fossdd** [comment]
You can revert the commit and re-commit it with the sign. Search on the Internet to search how to.

File: BUILDING.md [hunk 2]
```diff
@@ -121,20 +121,29 @@ cmake -GXCode -T buildsystem=1 -Duse_mad="off" -Duse_id3tag=off ../tenacity
 1. Clone Tenacity from the Tenacity GitHub project. 
   
     ```
-    $ git clone https://github.com/tenacityteam/tenacity/
+    $ git clone https://github.com/tenacityteam/tenacity/ && cd tenacity
```
Comment (fossdd) [inline BUILDING.md]: Can you cd in a new line?

**Review by fossdd — DISMISSED** [review]

File: BUILDING.md (hunk 2 above)
Comment (Songtech-0912) [reply BUILDING.md]: Sorry, I don't quite get what you mean. Do you mean this:

```bash
$ git clone https://github.com/tenacityteam/tenacity/
$ cd tenacity
```

Comment (Songtech-0912) [reply BUILDING.md]: I've separated it to a new line.

**Comment by Songtech-0912** [comment]
Both commits are now signed, the revert commit and the correction commit.

**Review by n0toose — DISMISSED** [review]

**Comment by n0toose** [comment]
Need additional confirmation here. The DCO test is still failing.

**Comment by Songtech-0912** [comment]
Ok, no problem. Take your time.

**Comment by purplesyringa** [comment]
Hi! I see that you signed-off the new commit, but the reverted commit is still in the history so DCO still does not pass. Can you please do something like this to replace the latest 3 commits with a single correct one?

```
git reset --soft HEAD~3  # revert 3 latest commits
git commit -s -m "Fixed instructions to enter repo before building on newline"  # commit new changes
git push --force-with-lease
```

Thank you!

**Comment by Songtech-0912** [comment]
Thank you for the advice! I have followed your instructions and force-pushed to git, that should've solved the unsigned past commit issue.

File: BUILDING.md [hunk 3]
```diff
@@ -122,19 +122,29 @@ cmake -GXCode -T buildsystem=1 -Duse_mad="off" -Duse_id3tag=off ../tenacity
...
    By default, Debug build will be configured. To change that, pass `-DCMAKE_BUILD_TYPE=Release` to CMake.
 
 3. Build Tenacity:
-   ```
+   ```bash
    $ make -j`nproc`
    ```
+   Note that this may slow your computer down quite a bit. To avoid this, you can use the alternate command:
+   ```bash
+   $ make -j`let numcores=$(nproc)-2 ; echo $numcores`
```
Comment (purplesyringa) [inline BUILDING.md]: What do you think about using `make -j$(($(nproc)-2))` instead? It's shorter and also works well in bash. I'm not sure whether it's too cryptic, what's your opinion?

Comment (Songtech-0912) [reply BUILDING.md]: Yeah, I agree, that would be a superior choice as long as there is an explanation of what the command means - I'll add that.

**Comment by Songtech-0912** [comment]
@imachug I've changed the `make` command to the one you suggested.

**Review by purplesyringa — APPROVED** [review]

**Comment by Semisol** [comment]
> Hi! I see that you signed-off the new commit, but the reverted commit is still in the history so DCO still does not pass. Can you please do something like this to replace the latest 3 commits with a single correct one?
> 
> ```
> git reset --soft HEAD~3  # revert 3 latest commits
> git commit -s -m "Fixed instructions to enter repo before building on newline"  # commit new changes
> git push --force-with-lease
> ```
> 
> Thank you!

You may also `git rebase HEAD~3 --signoff`.

**Comment by purplesyringa** [comment]
> You may also git rebase HEAD~3 --signoff.

Those the 3 commits were something like 'do change (no sign-off)', 'rollback previous commit', 'do change (with sign-off)' so I think this worked better but yeah nice to hear rebase supports signoff.

**Comment by Semisol** [comment]
> > You may also git rebase HEAD~3 --signoff.
> 
> Those the 3 commits were something like 'do change (no sign-off)', 'rollback previous commit', 'do change (with sign-off)' so I think this worked better but yeah nice to hear rebase supports signoff.

I tested, it does not double sign-off.

**Comment by Songtech-0912** [comment]
Checks look like they've all passed. Any other things that I may potentially need to change in the commit?

**Review by n0toose — APPROVED** [review]

**Comment by purplesyringa** [comment]
cc @fossdd (very sorry if rushing you)

**Comment by purplesyringa** [comment]
Thank you for your contribution!

**Pull request merged**